src/transport_ofi.c

```diff
@@ -87,7 +95,39 @@ static char                     myephostname[EPHOSTNAMELEN];
 #endif
 #ifdef ENABLE_THREADS
 shmem_internal_mutex_t          shmem_transport_ofi_lock;
-#endif
+#ifndef __APPLE__
+#ifdef HAVE_SYS_GETTID
+/* Need a syscall to gettid() because glibc doesn't provide a wrapper
+ * (see gettid manpage in the NOTES section): */
+static inline
+pid_t shmem_transport_ofi_gettid(void)
+{
+    return syscall(SYS_gettid);
+}
+#else
+static inline
+uint64_t shmem_transport_ofi_gettid(void)
+{
+    static uint64_t tid = 0;
```

Comment:
Add a comment here to the effect that we can't query TID, so we assume each query corresponds to a unique thread.